pinot-minion/src/main/java/org/apache/pinot/minion/taskfactory/TaskFactoryRegistry.java

```diff
@@ -67,45 +70,60 @@ public TaskFactoryRegistry(TaskExecutorFactoryRegistry taskExecutorFactoryRegist
             private final TaskConfig _taskConfig = context.getTaskConfig();
             private final PinotTaskExecutor _taskExecutor = taskExecutorFactory.create();
             private final MinionEventObserver _eventObserver = eventObserverFactory.create();
+            private final MinionMetrics _minionMetrics = MinionContext.getInstance().getMinionMetrics();
 
             @Override
             public TaskResult run() {
-              MinionMetrics minionMetrics = MinionContext.getInstance().getMinionMetrics();
+              HelixManager helixManager = context.getManager();
+              JobContext jobContext = TaskDriver.getJobContext(helixManager, context.getJobConfig().getJobId());
+              // jobContext.getStartTime() return the time in milliseconds of job being put into helix queue.
+              long jobInQueueTime = jobContext.getStartTime();
+              long jobDequeueTime = System.currentTimeMillis();
+              _minionMetrics.addPhaseTiming(taskType, MinionQueryPhase.TASK_QUEUEING, jobDequeueTime - jobInQueueTime);
+              try {
+                _minionMetrics.addValueToGlobalGauge(MinionGauge.NUMBER_OF_TASKS, 1L);
+                long startTimeMillis = System.currentTimeMillis();
+                TaskResult result = runInternal();
+                long timeSpentInMillis = System.currentTimeMillis() - startTimeMillis;
```

Comment:
these two lines should go into finally. even if task fails for whatever reason, it is time spent, and we should count it.

Reply:
done

Reply:
nit: rename it to sth like `executionTimeMs`